core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -586,15 +589,20 @@ class TransactionsTest extends IntegrationTestHarness {
       producer1.beginTransaction()
       val result =  producer1.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, null, "1", "5", willBeCommitted = false))
       val recordMetadata = result.get()
-      error(s"Missed a producer fenced exception when writing to ${recordMetadata.topic}-${recordMetadata.partition}. Grab the logs!!")
+      error(s"Missed an exception when writing to ${recordMetadata.topic}-${recordMetadata.partition}. Grab the logs!!")
       brokers.foreach { broker =>
         error(s"log dirs: ${broker.logManager.liveLogDirs.map(_.getAbsolutePath).head}")
       }
       fail("Should not be able to send messages from a fenced producer.")
     } catch {
-      case _: ProducerFencedException =>
-      case e: ExecutionException =>
-        assertTrue(e.getCause.isInstanceOf[ProducerFencedException])
+      case _: InvalidProducerEpochException =>
+      case e: ExecutionException => {
+        if (quorum == "zk") {
```

Comment:
Do we only use tv2 if we are using kraft? It is a bit tricky to understand the test here if we are using zk vs. kraft to determine which transaction version we are using.

Reply:
Updated the comments.